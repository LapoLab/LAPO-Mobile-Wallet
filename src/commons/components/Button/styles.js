// @flow
export default {
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonReady: {
    width: '80%',
    marginHorizontal: 25,
    justifyContent: 'center',
    alignItems: 'center',
    height: 64,
    backgroundColor: '#4183BD',
    borderRadius: 74,
    borderColor: '#BBCDEB',
    borderWidth: 8,
  },

  buttonReadyOtherBorder: {
    width: '80%',
    marginHorizontal: 25,
    justifyContent: 'center',
    alignItems: 'center',
    height: 64,
    backgroundColor: '#4183BD',
    borderRadius: 74,
    borderColor: 'rgb(42, 101, 117)',
    borderWidth: 8,
  },
  grayButton: {
    width: '80%',
    marginHorizontal: 25,
    justifyContent: 'center',
    alignItems: 'center',
    height: 64,
    backgroundColor: '#aeb4ba',
    borderRadius: 74,
    borderColor: '#d7dee5',
    borderWidth: 8,
  },
  textButtonReady: {
    fontFamily: 'Lato-Regular',
    fontSize: 18,
    color: 'white',
  },
};
