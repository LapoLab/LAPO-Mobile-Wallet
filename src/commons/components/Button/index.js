/**
 * @flow
 */

import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import styles from './styles';

type Props = {
  onPress: Function,
  title: string,
  status: 'active' | 'activeLight' | 'gray',
};
export default class Button extends Component<Props> {
  getStyleButton = (): Object => {
    const { status } = this.props;
    if (status === 'active') {
      return styles.buttonReady;
    }

    if (status === 'activeLight') {
      return styles.buttonReadyOtherBorder;
    }

    if (status === 'gray') {
      return styles.grayButton;
    }
    return {};
  };

  render() {
    const { onPress, title } = this.props;
    return (
      <View style={styles.buttonContainer}>
        <TouchableOpacity style={this.getStyleButton()} onPress={onPress}>
          <Text style={styles.textButtonReady}>{title}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
