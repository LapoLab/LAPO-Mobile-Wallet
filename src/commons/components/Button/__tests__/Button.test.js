// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('Button component', () => {
  test('renders snapshots default', () => {
    const fn = jest.fn();
    const tree = renderer.create(<Button title="Button" status="active"  onPress={fn} />); // eslint-disable-line
    const render = tree.toJSON();
    expect(render).toMatchSnapshot();
  });

  test('renders snapshots activeLight', () => {
    const fn = jest.fn();
    const tree = renderer.create(<Button title="Button" status="activeLight"  onPress={fn} />); // eslint-disable-line
    const render = tree.toJSON();
    expect(render).toMatchSnapshot();
  });

  test('renders snapshots gray ', () => {
    const fn = jest.fn();
    const tree = renderer.create(<Button title="Button" status="gray"  onPress={fn} />); // eslint-disable-line
    const render = tree.toJSON();
    expect(render).toMatchSnapshot();
  });
});
