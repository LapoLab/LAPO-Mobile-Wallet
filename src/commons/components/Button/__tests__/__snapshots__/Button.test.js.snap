// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders snapshots activeLight 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4183BD",
        "borderColor": "rgb(42, 101, 117)",
        "borderRadius": 74,
        "borderWidth": 8,
        "height": 64,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`Button component renders snapshots default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4183BD",
        "borderColor": "#BBCDEB",
        "borderRadius": 74,
        "borderWidth": 8,
        "height": 64,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`Button component renders snapshots gray  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#aeb4ba",
        "borderColor": "#d7dee5",
        "borderRadius": 74,
        "borderWidth": 8,
        "height": 64,
        "justifyContent": "center",
        "marginHorizontal": 25,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Lato-Regular",
          "fontSize": 18,
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;
